From b755ba6fd4772c015aace50647cf9f45924b6ef2 Mon Sep 17 00:00:00 2001
From: Lukas Bezdicka <lbezdick@redhat.com>
Date: Wed, 10 Sep 2014 19:34:11 +0200
Subject: [PATCH] Add manage_service feature

puppet-glance lacks of disabling service managing. This patch adds
$manage_service parameter to relevant classes.

Change-Id: I222b1a3318f5163f6ad1e39cbb8be10c440ab99f
Closes-bug: #1359823

Conflicts:
        manifests/api.pp
        manifests/registry.pp
---
 manifests/api.pp      | 15 +++++++++++----
 manifests/registry.pp | 34 +++++++++++++++++++++-------------
 2 files changed, 32 insertions(+), 17 deletions(-)

diff --git a/manifests/api.pp b/manifests/api.pp
index 9b68695..a5745ad 100644
--- a/manifests/api.pp
+++ b/manifests/api.pp
@@ -96,6 +96,10 @@
 #   (optional) User to authenticate as with keystone.
 #   Defaults to 'glance'.
 #
+# [*manage_service*]
+#   (optional) If Puppet should manage service startup / shutdown.
+#   Defaults to true.
+#
 # [*enabled*]
 #   (optional) Whether to enable services.
 #   Defaults to true.
@@ -181,6 +185,7 @@ class glance::api(
   $pipeline                 = 'keystone+cachemanagement',
   $keystone_tenant          = 'services',
   $keystone_user            = 'glance',
+  $manage_service           = true,
   $enabled                  = true,
   $use_syslog               = false,
   $log_facility             = 'LOG_USER',
@@ -423,10 +428,12 @@ class glance::api(
           '/etc/glance/glance-cache.conf']:
   }
 
-  if $enabled {
-    $service_ensure = 'running'
-  } else {
-    $service_ensure = 'stopped'
+  if $manage_service {
+    if $enabled {
+      $service_ensure = 'running'
+    } else {
+      $service_ensure = 'stopped'
+    }
   }
 
   service { 'glance-api':
diff --git a/manifests/registry.pp b/manifests/registry.pp
index bac8b71..da242e3 100644
--- a/manifests/registry.pp
+++ b/manifests/registry.pp
@@ -84,8 +84,13 @@
 #    (optional) Syslog facility to receive log lines.
 #    Defaults to LOG_USER.
 #
+#  [*manage_service*]
+#    (optional) If Puppet should manage service startup / shutdown.
+#    Defaults to true.
+#
 #  [*enabled*]
-#    (optional) Should the service be enabled. Defaults to true.
+#    (optional) Should the service be enabled.
+#    Defaults to true.
 #
 #  [*purge_config*]
 #    (optional) Whether to create only the specified config values in
@@ -128,6 +133,7 @@ class glance::registry(
   $pipeline              = 'keystone',
   $use_syslog            = false,
   $log_facility          = 'LOG_USER',
+  $manage_service        = true,
   $enabled               = true,
   $purge_config          = false,
   $cert_file             = false,
@@ -317,21 +323,23 @@ class glance::registry(
           '/etc/glance/glance-registry-paste.ini']:
   }
 
-  if $enabled {
 
-    Exec['glance-manage db_sync'] ~> Service['glance-registry']
+  if $manage_service {
+    if $enabled {
+      Exec['glance-manage db_sync'] ~> Service['glance-registry']
 
-    exec { 'glance-manage db_sync':
-      command     => $::glance::params::db_sync_command,
-      path        => '/usr/bin',
-      user        => 'glance',
-      refreshonly => true,
-      logoutput   => on_failure,
-      subscribe   => [Package[$glance::params::registry_package_name], File['/etc/glance/glance-registry.conf']],
+      exec { 'glance-manage db_sync':
+        command     => $::glance::params::db_sync_command,
+        path        => '/usr/bin',
+        user        => 'glance',
+        refreshonly => true,
+        logoutput   => on_failure,
+        subscribe   => [Package[$glance::params::registry_package_name], File['/etc/glance/glance-registry.conf']],
+      }
+      $service_ensure = 'running'
+    } else {
+      $service_ensure = 'stopped'
     }
-    $service_ensure = 'running'
-  } else {
-    $service_ensure = 'stopped'
   }
 
   service { 'glance-registry':
-- 
1.9.3

